src/dune_rules/pkg_rules.ml

```diff
@@ -621,6 +623,109 @@ module Substitute = struct
 end
 
 module Run_with_path = struct
+  module Output : sig
+    type t
+
+    val io : t -> Process.Io.output Process.Io.t
+
+    val v
+      :  ?accepted_exit_codes:int Predicate.t
+      -> ?pkg_name:Dune_lang.Package_name.t
+      -> string
+      -> t
+
+    val consume_and_print_error : t -> Display.t -> code:int -> loc:Loc.t -> unit
+  end = struct
+    type state =
+      | Open
+      | Closed
+
+    type t =
+      { pkg_name : Dune_pkg.Package_name.t option
+      ; fn : Dpath.t
```

Comment:
`fn` is a bit of a poor name, as it reminds me more of "function" than `filename`.